custom/icds/location_reassignment/download.py

```diff
@@ -121,48 +120,43 @@ def _add_validation(worksheet):
                 operation_data_validation.add(f"{letter}2:{letter}{worksheet.max_row}")
 
     def _create_rows(self):
-        def append_row(username):
-            row = deepcopy(location_content)
-            row[USERNAME_COLUMN] = username
-            row[NEW_USERNAME_COLUMN] = ''
-            row[OPERATION_COLUMN] = ''
-            rows[location_type].append(row)
-
         # location type code mapped to rows which is a list of dictionaries to pull headers later using keys
         rows = defaultdict(list)
         for location_id, location_details in self._location_details_by_location_id.items():
-            location_content = self._location_content(location_id)
             location_type = location_details['location_type']
             if location_details['assigned_users']:
                 for username in location_details['assigned_users']:
-                    append_row(username)
+                    rows[location_type].append(self._generate_row(username, location_type, location_id))
             else:
                 # set up one row for location that has no user assigned
-                append_row('')
+                rows[location_type].append(self._generate_row('', location_type, location_id))
         return rows
 
-    @memoized
-    def _location_content(self, location_id):
-        # memoized static content for each location
+    def _generate_row(self, username, location_type, location_id):
         location_details = self._location_details_by_location_id[location_id]
         location_parent_id = location_details['parent_location_id']
-        location_content = {
-            CURRENT_NAME: location_details['name'],
-            NEW_NAME: '',
+        row = {
             CURRENT_SITE_CODE_COLUMN: location_details['site_code'],
             NEW_SITE_CODE_COLUMN: '',
+            CURRENT_NAME: location_details['name'],
+            NEW_NAME: '',
+        }
+        row.update({
```

Comment:
Why is this a separate statement from the row above?

Reply:
@orangejenny kept it like that for the changes in the following commit, sorry that was confusing.

Reply:
@orangejenny kept it like that for the changes in the following commit, sorry that was confusing.